src/main.lua

```diff
@@ -3225,4 +3253,186 @@ addEventingHandler("paginatedDelegations", utils.hasMatchingTag("Action", "Pagin
 	})
 end)
 
+--- PRIMARY NAMES
+addEventingHandler("removePrimaryName", utils.hasMatchingTag("Action", ActionMap.RemovePrimaryNames), function(msg)
+	--- @param input table the message to check assertions and return the result
+	local checkAssertionsAndReturnResult = function(input)
```

Comment:
There's no need to parameterize the incoming message as `input` here and perform some of the checks below since you can just use block capture on the msg object.

Reply:
luacheck doesn't like overloaded variable names

Reply:
it blends - [24a6f10](https://github.com/ar-io/ar-io-network-process/pull/175/commits/24a6f1030bc06c3a1875c446318ff19d613ec6bd)